### PR TITLE
Correct implementation of Facebook-OAuth 

### LINKED ISSUE #5668: Facebook Login doesn't work
**Describe the bug**
When testing the website locally using API server also running locally, clicking on Facebook Login does not trigger anything. It doesn't show any pop-up or any error in logs or on the website.

**Expected behavior**
It should show a pop-up to ask permission for using our Facebook's info.

**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5668 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/helpers/third_party_auth.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def get_client_secret(cls):
     @classmethod
     def get_redirect_uri(cls):
         url = urllib.parse.urlparse(request.url)
-        fb_redirect_uri = url.scheme + '://' + url.netloc + '/fCallback'
+        fb_redirect_uri = 'http://localhost:4200/oauth/callback?provider=facebook'
```
Comment (CosmicCoder96) [inline app/api/helpers/third_party_auth.py]: what is this? Why would you hardcode localhost:4200 on the server?

Comment (prateekj117) [reply app/api/helpers/third_party_auth.py]: @CosmicCoder96 Just for testing right now.

File: app/api/helpers/third_party_auth.py [hunk 2]
```diff
@@ -59,8 +59,8 @@ def get_client_secret(cls):
 
     @classmethod
     def get_redirect_uri(cls):
-        url = urllib.parse.urlparse(request.url)
-        fb_redirect_uri = 'http://localhost:4200/oauth/callback?provider=facebook'
+        url = get_settings()['frontend_url'] + '/oauth/callback?provider=facebook'
```
Comment (iamareebjamal) [inline app/api/helpers/third_party_auth.py]: There's a util method called get_frontend_url. Use that. Also, remove WIP from title if you're done and want to get the PR reviewed. Mentors ignore reviewing PRs with WIP in title

File: app/api/helpers/third_party_auth.py [hunk 3]
```diff
@@ -59,8 +60,9 @@ def get_client_secret(cls):
 
     @classmethod
     def get_redirect_uri(cls):
-        url = urllib.parse.urlparse(request.url)
-        fb_redirect_uri = url.scheme + '://' + url.netloc + '/fCallback'
+        url = make_frontend_url(
+            '/oauth/callback?provider=facebook')
+        fb_redirect_uri = url
```
Comment (iamareebjamal) [inline app/api/helpers/third_party_auth.py]: Directly return url. New variable is redundant

**Comment by prateekj117** [comment]
@CosmicCoder96 Please review.

**Pull request merged**